common/djangoapps/student/models.py

```diff
@@ -123,7 +123,6 @@ class AnonymousUserId(models.Model):
     user = models.ForeignKey(User, db_index=True)
     anonymous_user_id = models.CharField(unique=True, max_length=32)
     course_id = CourseKeyField(db_index=True, max_length=255, blank=True)
-    unique_together = (user, course_id)
```

Comment:
@cpennington You are right, I didn't even notice this (and didn't think about why no migration was generated).  Thanks for merging!